playbooks/besman-criticality_score-0.0.1-playbook.sh

```diff
@@ -0,0 +1,147 @@
+
+    local flag=false
+    for var in "${var_array[@]}"; do
+        if [[ ! -v $var ]]; then
+
+            # read -rp "Enter value for $var:" value #remove
+            # export "$var"="$value" #remove
+            __besman_echo_yellow "$var is not set" #uncomment
+            __besman_echo_no_colour ""             #uncomment
+            flag=true                              #uncomment
+        fi
+
+    done
+
+    local dir_array=("BESMAN_ASSESSMENT_DATASTORE_DIR")
+    for dir in "${dir_array[@]}"; do
+        # Get the value of the variable with the name stored in $dir
+        dir_path="${!dir}"
+
+        if [[ ! -d $dir_path ]]; then
+            __besman_echo_red "Could not find $dir_path"
+            flag=true
+        fi
+
+    done
+
+   # [[ ! -f $BESMAN_TOOL_PATH/$ASSESSMENT_TOOL_NAME ]] && __besman_echo_red "Could not find artifact @ $BESMAN_TOOL_PATH/$ASSESSMENT_TOOL_NAME" && flag=true
+
+    if [[ $flag == true ]]; then
+        return 1
+    else
+        export CRITICALITY_SCORE_PATH="$BESMAN_ASSESSMENT_DATASTORE_DIR/$BESMAN_ARTIFACT_NAME/$BESMAN_ARTIFACT_VERSION/criticality_score"
+        export DETAILED_REPORT_PATH="$CRITICALITY_SCORE_PATH/$BESMAN_ARTIFACT_NAME-$BESMAN_ARTIFACT_VERSION-criticality_score-report.json"
+        mkdir -p "$CRITICALITY_SCORE_PATH"
+        export OSAR_PATH="$BESMAN_ASSESSMENT_DATASTORE_DIR/osar/$BESMAN_ARTIFACT_NAME-$BESMAN_ARTIFACT_VERSION-OSAR.json"
+        __besman_fetch_steps_file "$steps_file_name" || return 1
+        return 0
+
+    fi
+
+}
+
+function __besman_execute() {
+    local duration
+    __besman_echo_yellow "Launching steps file"
+
+    SECONDS=0
+    . "$BESMAN_STEPS_FILE_PATH"
+    duration=$SECONDS
+
+    export EXECUTION_DURATION=$duration
+    if [[ $CRITICALITY_SCORE_RESULT == 1 ]]; then
+
+        export PLAYBOOK_EXECUTION_STATUS=failure
+        return 1
+
+    else
+        export PLAYBOOK_EXECUTION_STATUS=success
+        return 0
+    fi
+
+}
+
+function __besman_prepare() {
+    __besman_echo_white "preparing data"
+    EXECUTION_TIMESTAMP=$(date)
+    export EXECUTION_TIMESTAMP
+
+    __besman_generate_osar
+
+}
+
+function __besman_publish() {
+    __besman_echo_yellow "Pushing to datastores"
+    # push code to remote datastore
+    cd "$BESMAN_ASSESSMENT_DATASTORE_DIR"
+
+    git add "$DETAILED_REPORT_PATH" "$OSAR_PATH"
+    git commit -m "Added osar and detailed report"
+    git push origin main
+}
+
+function __besman_cleanup() {
+    local var_array=("ASSESSMENT_TOOL_NAME" "ASSESSMENT_TOOL_TYPE" "ASSESSMENT_TOOL_PLAYBOOK" "ASSESSMENT_TOOL_VERSION" "OSAR_PATH" "CRITICALITY_SCORE_PATH" "DETAILED_REPORT_PATH")
+
+    :'for var in "${var_array[@]}"; do
```

Comment:
@NeerajK007 This is throwing error,

![image](https://github.com/Be-Secure/besecure-playbooks-store/assets/58501088/fae4c1d7-37be-4a25-92d0-a7c7cf259bb2)